### DISCUSSION TITLE
UDM SE - KPN IPTV (Xperiabox 12 - Sagemcom)

### DISCUSSION THREAD

**Discussion by eradicationx** [fabianishere/udm-iptv#172] [Q&A]
Hello everyone and many thanks in advance.

(Warning in advance my linux knowledge is pretty much zero)

I recently got a UDM SE (2 weeks ago as of today) and quite happy with it as after some years I finally decided to do a complete overhaul of the network at home as well.

Currently I have my KPN-IPTV directly linked to my UDM SE and getting the infamous stuttering / blue screen errors after roughly a few seconds and then I came across this glorious script made possible by Fabian.
Unfortunately even while following the steps I can't get it to work.

Reading into it further it could be related to the Sagemcom box I am behind but I'd like to make sure that I am infact able to run IPTV traffic through the UDM SE before giving up and leaving my IPTV's connected directly to the Sagemcom box.

I set up a separate VLAN tried it with IGMP Snooping Enabled and Disabled, same with Multicast DNS but to no success whatsoever.
Running the script from Fabian I selected the proper vlan to run everything through but either I have missed something entirely or just have no freaking idea as to what's going on so hoping some of you fine people is able to assist me.

Here's my journal:
Sep 18 21:54:46 XXX systemd[1]: udm-iptv.service: Main process exited, code=exited, status=255/EXCEPTION
Sep 18 21:54:46 XXX systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Sep 18 21:54:46 XXX systemd[1]: udm-iptv.service: Consumed 112ms CPU time.
Sep 18 21:54:51 XXX systemd[1]: udm-iptv.service: Service RestartSec=5s expired, scheduling restart.
Sep 18 21:54:51 XXX systemd[1]: udm-iptv.service: Scheduled restart job, restart counter is at 40.
Sep 18 21:54:51 XXX systemd[1]: Stopped IPTV support for the UniFi Dream Machine.
Sep 18 21:54:51 XXX systemd[1]: udm-iptv.service: Consumed 112ms CPU time.
Sep 18 21:54:51 XXX systemd[1]: Started IPTV support for the UniFi Dream Machine.
Sep 18 21:54:51 XXX udm-iptv[4879]: Device iptv already exists.. Deleting device
Sep 18 21:54:51 XXX udm-iptv[4879]: Obtaining IP address for VLAN interface
Sep 18 21:54:51 XXX udm-iptv[4879]: udhcpc (v1.22.1) started
Sep 18 21:54:51 XXX udm-iptv[4879]: Sending discover...
Sep 18 21:54:54 XXX udm-iptv[4879]: Sending discover...
Sep 18 21:54:58 XXX udm-iptv[4879]: Sending discover...
Sep 18 21:55:01 XXX udm-iptv[4879]: udhcpc failed to get a DHCP lease
Sep 18 21:55:01 XXX udm-iptv[4879]: No lease, forking to background
Sep 18 21:55:01 XXX udm-iptv[4879]: NATing IPTV network ranges (if necessary)
Sep 18 21:55:01 XXX udm-iptv[4879]: Setting up igmpproxy
Sep 18 21:55:01 XXX udm-iptv[4879]: Starting igmpproxy
Sep 18 21:55:01 XXX udm-iptv[4879]: adding VIF, Ix 0 Fl 0x0 IP 0x0104a8c0 br4, Threshold: 1, Ratelimit: 0
Sep 18 21:55:01 XXX udm-iptv[4879]: There must be at least 1 Vif as upstream.
Sep 18 21:55:01 XXX systemd[1]: udm-iptv.service: Main process exited, code=exited, status=255/EXCEPTION
Sep 18 21:55:01 XXX systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Sep 18 21:55:01 XXX systemd[1]: udm-iptv.service: Consumed 136ms CPU time.
Sep 18 21:55:06 XXX systemd[1]: udm-iptv.service: Service RestartSec=5s expired, scheduling restart.
Sep 18 21:55:06 XXX systemd[1]: udm-iptv.service: Scheduled restart job, restart counter is at 41.
Sep 18 21:55:06 XXX systemd[1]: Stopped IPTV support for the UniFi Dream Machine.
Sep 18 21:55:06 XXX systemd[1]: udm-iptv.service: Consumed 136ms CPU time.
Sep 18 21:55:06 XXX systemd[1]: Started IPTV support for the UniFi Dream Machine.
Sep 18 21:55:06 XXX udm-iptv[5740]: Device iptv already exists.. Deleting device
Sep 18 21:55:06 XXX udm-iptv[5740]: Obtaining IP address for VLAN interface
Sep 18 21:55:06 XXX udm-iptv[5740]: udhcpc (v1.22.1) started
Sep 18 21:55:06 XXX udm-iptv[5740]: Sending discover...
Sep 18 21:55:09 XXX udm-iptv[5740]: Sending discover...
Sep 18 21:55:12 XXX udm-iptv[5740]: Sending discover...

udm-iptv-diag:
=== Configuration ===
WAN Interface: eth8
WAN VLAN: 4 (dev iptv)
WAN DHCP: true (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 213.75.0.0/16  217.166.0.0/16
LAN Interfaces: br4
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
=== Service Logs ===
Sep 18 22:07:47 XXX systemd[1]: udm-iptv.service: Consumed 129ms CPU time.
Sep 18 22:07:52 XXX systemd[1]: udm-iptv.service: Service RestartSec=5s expired, scheduling restart.
Sep 18 22:07:52 XXX systemd[1]: udm-iptv.service: Scheduled restart job, restart counter is at 93.
Sep 18 22:07:52 XXX systemd[1]: Stopped IPTV support for the UniFi Dream Machine.
Sep 18 22:07:52 XXX systemd[1]: udm-iptv.service: Consumed 129ms CPU time.
Sep 18 22:07:52 XXX systemd[1]: Started IPTV support for the UniFi Dream Machine.
Sep 18 22:07:52 XXX udm-iptv[13551]: Device iptv already exists.. Deleting device
Sep 18 22:07:52 XXX udm-iptv[13551]: Obtaining IP address for VLAN interface
Sep 18 22:07:52 XXX udm-iptv[13551]: udhcpc (v1.22.1) started
Sep 18 22:07:53 XXX udm-iptv[13551]: Sending discover...

ip -4 addr show dev iptv doesn't show ip, without the -4 it shows:
root@XXX:~# ip addr show dev iptv
140: iptv@eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether d0:21:f9:88:6a:49 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::d221:f9ff:fe88:6a49/64 scope link
       valid_lft forever preferred_lft forever

I am running the latest (Official) network and Unifi OS version
OS: 2.5.11
Network: 7.2.94

I could simply attach it to my regular KPN box and get a simple switch, I have however taken this route and am quite stubborn and wait to see it through. :P

**Comment by fabianishere** [comment]
What is the Sagecom box you are talking about? If it is another router in front of your UDM SE, it must also route the IPTV traffic (with a multicast proxy).

**Reply by eradicationx** [reply]
It's the experiabox 12
[experiabox 12](https://uploads-eu-west-1.insided.com/kpn-nl/attachment/862a40f1-e6ff-4363-825e-72e4bfbc919c.png)

Fibre comes into the house, into that, into my UDM SE.
(Router/modem, doesn't have option for multicast proxy that I am able to edit :P plus connecting the IPTV directly to that works flawlessly)

**Reply by fabianishere** [reply]
It won’t work with the default settings if the UDM SE is connected via de Experiabox. Does it work if you set WAN VLAN to 0 (IPTV traffic not carried over a separate VLAN)

**Reply by eradicationx** [reply]
Negative, if I set VLAN id to 0 everything drops. 
Also forces me to enter QoS tag as it doesn't accept "0"
Regardless of what VLAN tag I use I won't get any connection, this is (I am assuming) being caused by the fact the UDM is not directly connected to Fibre.

I'll be looking into trying it with a direct connection once I've shipped everyone in the house to a resort :D

**Reply by fabianishere** [reply]
What do you mean by everything drops? And how do you set the VLAN ID to 0 (do you use the installer of this script for this)?

**Reply by eradicationx** [reply]
My entire connection drops and is no longer accessible until I change it back (within the UDM) that's where I changed the VLAN ID.

Following the steps from your IPTV on Unifi OS guide I was not able to set it to 0 (I am severely lacking know how outside of the steps in the guide when it comes to linux)

If you can throw any commands my way feel free to. :)

**Reply by fabianishere** [reply]
I meant changing the WAN VLAN to 0 in `/etc/udm-iptv.conf`. The property is called `IPTV_WAN_VLAN`.

**Reply by eradicationx** [reply]
Ah, my bad!

I get the same bad connection but now I at least see some stuff in udm-iptv-diag:
=== Configuration ===
WAN Interface: eth8
WAN VLAN: 0 (dev iptv)
WAN DHCP: true (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 213.75.0.0/16  217.166.0.0/16
LAN Interfaces: br4
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
11: eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    inet 192.168.2.6/24 scope global dynamic eth8
       valid_lft 52754sec preferred_lft 52754sec
192.168.2.0/24 proto kernel scope link src 192.168.2.6
=== Service Logs ===
Sep 20 18:42:28 XXX udm-iptv[20475]: RECV V2 member report   from 192.168.4.183   to 224.3.2.6
Sep 20 18:42:28 XXX udm-iptv[20475]: Updated route entry for 224.3.2.6 on VIF #1
Sep 20 18:42:28 XXX udm-iptv[20475]: Adding MFC: 213.75.167.58 -> 224.3.2.6, InpVIf: 0
Sep 20 18:42:28 XXX udm-iptv[20475]: Joining group 224.3.2.6 on interface eth8
Sep 20 18:42:28 XXX udm-iptv[20475]: RECV V2 member report   from 192.168.4.183   to 224.3.2.6
Sep 20 18:42:28 XXX udm-iptv[20475]: Updated route entry for 224.3.2.6 on VIF #1
Sep 20 18:42:28 XXX udm-iptv[20475]: Adding MFC: 213.75.167.58 -> 224.3.2.6, InpVIf: 0
Sep 20 18:42:32 XXX udm-iptv[20475]: RECV V2 member report   from 192.168.4.183   to 224.3.2.6
Sep 20 18:42:32 XXX udm-iptv[20475]: Updated route entry for 224.3.2.6 on VIF #1
Sep 20 18:42:32 XXX udm-iptv[20475]: Adding MFC: 213.75.167.58 -> 224.3.2.6, InpVIf: 0
root@XXX:~#


On my TV it now displays the infamous stb-nmc-400
(I did restart the service)

**Reply by eradicationx** [reply]
figured I'd add new journal as well:
Sep 20 18:44:58 XXX udm-iptv[20475]: Joining group 224.3.2.6 on interface eth8
Sep 20 18:44:59 XXX udm-iptv[20475]: The source address 192.168.4.1 for group 233.89.188.1
Sep 20 18:45:05 XXX udm-iptv[20475]: RECV Membership query   from 192.168.4.1     to 224.0.
Sep 20 18:45:06 XXX udm-iptv[20475]: RECV V2 member report   from 192.168.4.183   to 224.3.
Sep 20 18:45:06 XXX udm-iptv[20475]: Updated route entry for 224.3.2.6 on VIF #1
Sep 20 18:45:06 XXX udm-iptv[20475]: Adding MFC: 213.75.167.58 -> 224.3.2.6, InpVIf: 0
Sep 20 18:45:07 XXX udm-iptv[20475]: RECV V2 member report   from 192.168.4.183   to 225.0.
Sep 20 18:45:07 XXX udm-iptv[20475]: Updated route entry for 225.0.71.1 on VIF #1
Sep 20 18:45:08 XXX udm-iptv[20475]: The IGMP message was local multicast. Ignoring.
Sep 20 18:45:08 XXX udm-iptv[20475]: RECV Leave message      from 192.168.4.183   to 224.0.
Sep 20 18:45:08 XXX udm-iptv[20475]: Leaving group 224.3.2.6 on interface eth8
Sep 20 18:45:08 XXX udm-iptv[20475]: RECV Membership query   from 192.168.4.1     to 224.3.
Sep 20 18:45:08 XXX udm-iptv[20475]: RECV V2 member report   from 192.168.4.183   to 224.3.
Sep 20 18:45:08 XXX udm-iptv[20475]: Updated route entry for 224.3.2.6 on VIF #1
Sep 20 18:45:08 XXX udm-iptv[20475]: Adding MFC: 213.75.167.58 -> 224.3.2.6, InpVIf: 0
Sep 20 18:45:08 XXX udm-iptv[20475]: Joining group 224.3.2.6 on interface eth8
Sep 20 18:45:09 XXX udm-iptv[20475]: The IGMP message was local multicast. Ignoring.

**Reply by eradicationx** [reply]
funny enough if I go to a channel that allows it to be paused for a few seconds it then works flawlessly afterwards.. (unless I switch channels again) shrug

that method didn't work before but still finding it strange..

**Reply by fabianishere** [reply]
Is your IPTV receiver also connected to VLAN 4 on your local network?

**Reply by eradicationx** [reply]
Yeah it is, it's the 192.168.4.183 IP address, on my UDM connected to port 6 which is the 192.168.4 vlan

**Reply by fabianishere** [reply]
Perhaps you could try adding a switch between the UDM and the IPTV receiver. There have been issues with IGMP snooping on the UDM.

**Reply by eradicationx** [reply]
Ahh I see.

I do have one of those dumb switches around which was like 25€ but had the same issue there (shocking) makes sense I guess considering there's 0 management on it.

Was looking to get one of the Ubiquiti switches so I'll try that out.
Thanks for everything thus far!

**Reply by eradicationx** [reply]
update:
Got a small TP link switch (TL-SG105E) result was the same.
Connecting my UDM SE directly to the fiber and then running: dpkg-reconfigure -p medium udm-iptv once again it works flawlessly.

Safe to safe the Experiabox 12 is a b*tch..

I'd like to thank you very much for your efforts fabian, finally feel at ease knowing it wasn't anything wrong on my end. :P
